### PR TITLE
updated to new KNP

### PR DESCRIPTION
**Pull request by wadeking98**

### CODE REVIEW

File: openshift/templates/nsp/build.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        name: pods-to-external-network
+        env: ${ENV_NAME}
+    spec:
+      description: |
+        Allow the builds to access the internet.
+        This only needs to be specified once per environment.
+      source:
+        - - $namespace=${NAMESPACE_NAME}-${ENV_NAME}
+      destination:
+        - - ext:name=internet-build-resources
```
Comment (esune) [inline openshift/templates/nsp/build.yaml:54]: Outdated: needs to be in the new KNP format

**Review by esune — CHANGES_REQUESTED** [review]
A couple of tweaks are required.

Additionally, I would suggest the client base build and the server build are updated to use Artifactory: the server is already setup correctly, you only need to add the artifactory prefix to the `SOURCE_IMAGE_NAME` parameter.

The client build needs to be updated to follow the same pattern.

File: openshift/templates/nsp/build.yaml (hunk 1 above)
Comment (wadeking98) [reply openshift/templates/nsp/build.yaml]: changed to recommended exaple by Wade Barnes

File: openshift/templates/client/client-deploy.dev.param [hunk 2]
```diff
@@ -8,14 +8,14 @@
...
-# NAMESPACE_NAME=devex-von
+# IMAGE_NAMESPACE=a99fd4-tools
+# NAMESPACE_NAME=a99fd4
 # NGINX_CONFIG_MOUNT_PATH=/etc/caddy/
 # NGINX_CONFIG_FILE_NAME=Caddyfile
 # APP_CONFIG_MAP_NAME=dsc-chat-client-config
 # APP_CONFIG_MOUNT_PATH=/srv/assets/
 # APP_CONFIG_FILE_NAME=config.json
-APPLICATION_DOMAIN=pctftest-dev.vonx.io
+APPLICATION_DOMAIN=pctftest-dev.apps.silver.devops.gov.bc.ca
```
Comment (esune) [inline openshift/templates/client/client-deploy.dev.param:18]: The `vonx.io` URLs should be retained for dev and test as well

Comment (wadeking98) [reply openshift/templates/client/client-deploy.dev.param]: reverted to vonx.io

File: openshift/templates/server/server-build.param [hunk 3]
```diff
@@ -6,13 +6,13 @@
...
 APP_GROUP=dsc-chat-runtime
-NAMESPACE_NAME=devex-von
+NAMESPACE_NAME=a99fd4
 ENV_NAME=tools
 GIT_REPO_URL=https://github.com/bcgov/issuer-kit-demo-verifier-chat.git
 GIT_REF=main
 SOURCE_CONTEXT_DIR=chat-server
 SOURCE_IMAGE_KIND=DockerImage
-SOURCE_IMAGE_NAME=centos/nodejs-12-centos7
+SOURCE_IMAGE_NAME=docker-remote.artifacts.developer.gov.bc.ca/centos/nodejs-12-centos7
```
Comment (WadeBarnes) [inline openshift/templates/server/server-build.param:15]: There is a feely available UBI replacement for this; `registry.access.redhat.com/ubi8/nodejs-12`

```suggestion
SOURCE_IMAGE_NAME=registry.access.redhat.com/ubi8/nodejs-12
```

File: openshift/templates/server/server-build.yaml [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    displayName: Source Image Kind
+    required: true
+    description: The 'kind' (type) of the  source image; typically ImageStreamTag,
+      or DockerImage.
+    value: DockerImage
+  - name: SOURCE_IMAGE_NAME
+    displayName: Source Image Name
+    required: true
+    description: The name of the source image.
+    value: docker-remote.artifacts.developer.gov.bc.ca/centos/nodejs-12-centos7
```
Comment (WadeBarnes) [inline openshift/templates/server/server-build.yaml:124]:
```suggestion
    value: registry.access.redhat.com/ubi8/nodejs-12
```

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
See comments and suggestions.

**Comment by wadeking98** [comment]
dco complaining about a 3 way merge, going to close and re open PR

**Pull request closed without merging**